doc/cephfs/nfs.rst

```diff
@@ -0,0 +1,157 @@
+===
+NFS
+===
+
+CephFS namespaces can be exported over NFS protocol using the
+`NFS-Ganesha NFS server <https://github.com/nfs-ganesha/nfs-ganesha/wiki>`_.
+
+Requirements
+============
+
+-  Ceph filesystem (preferably latest luminous or later versions)
+-  'nfs-ganesha' and 'nfs-ganesha-ceph' packages (preferably the latest
+   ganesha v2.5 stable or later stable versions)
+
+   +  RPMs and packages for xenial available at http://download.ceph.com/nfs-ganesha/
+
+-  NFS-Ganesha server host connected to the Ceph public network
+
+Configuring NFS-Ganesha to export CephFS
+========================================
+
+NFS-Ganesha provides a File System Abstraction Layer (FSAL) to plug in different
+storage backends. `Ceph FSAL <https://github.com/nfs-ganesha/nfs-ganesha/tree/next/src/FSAL/FSAL_CEPH>`_
+is the plugin FSAL for CephFS. For each NFS-Ganesha export, Ceph FSAL uses a
+libcephfs client, user-space CephFS client, to mount the CephFS path that
+NFS-Ganesha exports.
+
+Setting up NFS-Ganesha with CephFS, involves setting up NFS-Ganesha's
+configuration file, and also setting up a Ceph configuration file and cephx
+access credentials for the Ceph clients created by NFS-Ganesha to access
+CephFS.
+
+
+NFS-Ganesha configuration
+-------------------------
+
+A minimal ganesha.conf for configuring NFS-Ganesha over CephFS includes a
+EXPORT block and a CEPH block. Following is an example::
+
+    EXPORT
+    {
+        Export_ID = 100;
+        Path = "/"
+        Pseudo = "/cephfs/";
+        Protocols = 4;
+        # Squash = Root_Squash;
+
+        FSAL
+        {
+            Name = Ceph;
+            # User_Id = "{cephx user ID}";
+            # Secret_Access_Key = "{cephx secret key}";
+        }
+
+        CLIENT
+        {
+		    Clients = 192.168.0.10,192.168.1.0/8;
+		    Access_Type = RW ;
+        }
+    }
+
+    # Config block for Ceph FSAL
+    CEPH
+    {
+	    # Ceph_Conf = /etc/ceph/ceph.conf;
+
+	    # umask = 0;
+    }
+
+EXPORT {} options
+
+- ``Export_ID`` must be a integer value.
```

Comment:
@jtlayton can you elaborate on the significance of the export ID? Does it need to be unique across a ganesha cluster?